openedx/core/djangoapps/system_wide_roles/constants.py

```diff
@@ -3,3 +3,6 @@
 """
 
 STUDENT_SUPPORT_ADMIN_ROLE = 'student_support_admin'
+
+# Role needed to access support tools MFE
+SUPPORT_TOOLS_ADMIN_ROLE = "support_tools_admin"
```

Comment:
How does this new role differ from the role above?

@waheedahmed @brittneyexline @cpennington Can you provide context on the `STUDENT_SUPPORT_ADMIN_ROLE`? I do not see any docs that explain the intention of the scope of that role.

Reply:
Also, we need to establish a process for adding system-wide roles. It can easily get out of hand without intentionality in understanding the scope of each role, how we namespace it, etc.

Reply:
@nasthagiri Revoking admin permissions in e-commerce under PROD-123 broke the support refund flows and it was added to give specific support members access to the orders/refund tools in e-commerce where they can successfully process the refunds. I guess we could use this system-wide role here as well but it will give every support member access to orders and refund tables in e-commerce and I'm not sure if this is the desired outcome here.

Reply:
FYI - we're having an internal conversation on this current PR at https://edx-internal.slack.com/archives/CG7FM3BLY/p1612881916312700

Reply:
We need to consider the long-term sustainability of system-wide roles.

**What failure looks like:**

- We have a laundry list of system-wide roles created by each team for their own domain needs.
- Roles are too fine-grained and become unmanageable in a large platform.
- Administration of assigning users to roles becomes too complex.

**What success looks like:**

- We have a handful (less than 12) system-wide roles.
- They are clearly defined and tend to be mutually exclusive.
- They are easy to administer and it’s also clear from a development perspective of what the roles are intended for.
- Each microservice defines its own relationship between the system-wide roles and its service-specific roles.

Reply:
**Requirement:** Securely enforcing access needs to happen on the backend.
**Rationale:** Users can always change their frontend code in their own browsers and bypass any security checks you put in the frontend.

So, the backend implementation needs to do the access check in order for the implementation to be hacker-proof.

Now, the frontend code can do the access check in a few ways.

1. Gracefully handle any access control issues raised by the backend. Ensure the backend API provides adequate error information that can be presented to the user.
2. If the frontend wants to preemptively disable a button for usability reasons, the backend API can provide information on what the user is capable of doing. For example, the API can return boolean values for: can_adjust_grade, can_view_history, etc.
3. _(We haven’t done this in the platform yet..)_ Have a separate permission-specific backend API for the service you’re going to call eventually. The frontend would call this API first and cache the user’s permissions (maybe in local-storage). However, it will need to handle race conditions where the user’s permissions have changed on the backend.

Reply:
> give specific support members access to the orders/refund tools in e-commerce where they can successfully process the refunds

@waheedahmed That sounds like a need for a service-specific role, not a system-wide role.  Does any service other than e-commerce need to check for that role?

Reply:
@nasthagiri Yes, we have created a service-specific role [ORDER_MANAGER_ROLE in e-commerce service](https://github.com/edx/ecommerce/blob/2c4f5290a5da82c519c9154957c594fe8a445658/ecommerce/settings/base.py#L671) which then maps to this system-wide role.

Reply:
@nasthagiri Can this be moved ahead or are there still some concerns around the role being added?